core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java

```diff
@@ -121,7 +121,7 @@ public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long
       }
       throw new IOException(e);
     } finally {
-      if (channel != null && listener != null) {
+      if (channel != null && listener != null && channel.isActive()) {
```

Comment:
Discussed in person. In this case, the server should be crashed. I don't know what is the convention in Alluxio codebase. if we prefer explicit precondtion check. i can do that.